x/programs/rust/examples/single_nft/src/metadata.rs

```diff
@@ -0,0 +1,29 @@
+//! NFT schema
+//! See https://nftschool.dev/reference/metadata-schemas/#ethereum-and-evm-compatible-chains for more information
+//! on the ERC-721 NFT metadata schema.
+
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Nft {
+    pub symbol: String,
+    pub name: String,
+    pub uri: String,
+}
+
+impl Nft {
+    pub fn with_symbol(mut self, symbol: String) -> Self {
+        self.symbol = symbol;
+        self
+    }
+
+    pub fn with_name(mut self, name: String) -> Self {
+        self.name = name;
+        self
+    }
+
+    pub fn with_uri(mut self, uri: String) -> Self {
+        self.uri = uri;
+        self
+    }
+}
```

Comment:
Not sure what happened to my previous comment, but I really don't think it's worth having all this extra code with a builder pattern for an example. You could easily put the `Nft` type in the `lib.rs` file so that when someone goes to read the example, they see everything in one place.

Reply:
It's gone now